### PR TITLE
Add dependabot to keep GitHub Actions up to date

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

The dependabot workflow will open PRs to update versions of GitHub
Actions.

References:

1. https://docs.github.com/en/github/administering-a-repository/keeping-your-actions-up-to-date-with-dependabot
2. https://docs.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
See the PRs (https://github.com/seisman/pygmt/pulls) opened by dependabot in my own fork.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Set update schedule for GitHub Actions
+
+version: 2
+updates:
+
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      # Check for updates to GitHub Actions every weekday
+      interval: "daily"
```
Comment (weiji14) [inline .github/dependabot.yml:10]:
```suggestion
      interval: "weekly"
      day: "tuesday"
```
Checking weekly should suffice.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    # Specify labels for pull requests
+    labels:
+      - "maintenance"
```
Comment (weiji14) [inline .github/dependabot.yml:13]:
```suggestion
      - "maintenance"
      - "skip-changelog"
```
Maybe add skip-changelog as well?

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+version: 2
+updates:
+
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      # Check for updates to GitHub Actions every weekday
+      interval: "daily"
+    # Specify labels for pull requests
```
Comment (weiji14) [inline .github/dependabot.yml:11]:
```suggestion
    # Allow up to 2 open pull requests for pip dependencies
    open-pull-requests-limit: 2
    # Specify labels for pull requests
```
Default number of open dependabot Pull Requests is 5 (see https://docs.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates#open-pull-requests-limit) which might be a lot. I tend to stick with 2 on my other repos, but we can negotiate a bit on what number to go with.

**Review by weiji14 — COMMENTED** [review]
Just a few suggestions to tone down on how spammy dependabot can get, speaking from experience :slightly_smiling_face:

**Review by weiji14 — APPROVED** [review]

**Pull request merged**